js/core/utils/data.js

```diff
@@ -191,7 +193,11 @@ export const toComparable = function(value, caseSensitive) {
     }
 
     if(!caseSensitive && typeof value === 'string') {
-        return value.toLowerCase();
+        if(options?.collateOptions?.sensitivity === 'base') {
```

Comment:
Question.

Is it ok, that we apply sensitivity: 'base' only in case insensitive comparison?

Reply:
yep, Let it be for now